src/inference/src/threading/ie_istreams_executor.cpp

```diff
@@ -241,6 +334,89 @@ Parameter IStreamsExecutor::Config::GetConfig(const std::string& key) const {
     return {};
 }
 
+int IStreamsExecutor::Config::LimitHybridStreams(const int num_streams, const int num_threads) {
+    const int base_streams_total = parallel_get_max_threads();
+
+    // limit streams to reasonable range
+    int streams = num_streams > base_streams_total ? base_streams_total : std::max(1, num_streams);
```

Comment:
What's the motivation behind that? Why we need to **implicitly** adjust number of streams?

Reply:
Will move limitation logic to following PR and benefit to all affinities.